src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs

```diff
@@ -250,17 +250,34 @@ private void WriteClientMethods(CodeWriter writer, ResourceOperation resourceOpe
             // 3. Listing children (might be resource or not) -> on the operations
 
             // write rest of the methods
+            var resourceContainer = context.Library.GetResourceContainer(resourceOperation.OperationGroup);
             foreach (var clientMethod in resourceOperation.Methods)
             {
+                var isMethodExistInContainer = false;
```

Comment:
This logic should be in the property returned by resourceOperations otherwise any time we want the list of all the methods unique to the operations vs container we have to perform the same logic you have here for isMethodExistInContainer